Sources/Sentry/SentryProfiler.mm

```diff
@@ -258,13 +288,11 @@ - (SentryEnvelopeItem *)buildEnvelopeItemForTransaction:(SentryTransaction *)tra
     profile[@"device_is_emulator"] = @(isSimulatorBuild());
     profile[@"device_physical_memory_bytes"] =
         [@(NSProcessInfo.processInfo.physicalMemory) stringValue];
-    profile[@"environment"] = hub.scope.environmentString ?: hub.getClient.options.environment ?: kSentryDefaultEnvironment;
-    profile[@"platform"] = transaction.platform;
-    profile[@"transaction_id"] = transaction.eventId.sentryIdString;
-    profile[@"trace_id"] = transaction.trace.context.traceId.sentryIdString;
-    profile[@"profile_id"] = [[SentryId alloc] init].sentryIdString;
-    profile[@"transaction_name"] = transaction.transaction;
-    profile[@"duration_ns"] = [@(getDurationNs(_startTimestamp, getAbsoluteTime())) stringValue];
+    const auto profileID = [[SentryId alloc] init];
+    profile[@"profile_id"] = profileID.sentryIdString;
+    const auto profileDuration = getDurationNs(_startTimestamp, _endTimestamp);
+    profile[@"duration_ns"] = [@(profileDuration) stringValue];
+    profile[@"truncation_reason"] = profilerStopReasonName(stopReason);
```

Comment:
`m`: We have two different names for the same concept. If we call it `truncation_reason` in the protocol, I would suggest renaming this and the enum to 
```suggestion
    profile[@"truncation_reason"] = profilerTruncationReasonName(stopReason);
```